edx/frontend-app-payment
# REV-2217: add state list for India

## edX Internal Developers are expected to [release and monitor their changes](https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories)

## Description
REV-2217: when a user selects India as the country on checkout, require the state/region field to be filled out in order to submit the order.

This field is automatically required when the country has states populated in countryStatesMap, so I've populated this list using: 
states from here: https://en.wikipedia.org/wiki/List_of_state_and_union_territory_capitals_in_India
abbreviations from here: https://kb.bullseyelocations.com/article/60-india-state-codes
(note: the second list is bigger because it also includes union territories, which we don't want to include

## Supporting information
https://openedx.atlassian.net/browse/REV-2217

## Testing instructions
I confirmed: 
- when India is selected, state is required
- when India is selected, the state dropdown is displayed matching the wiki list
- when a different county (not the other two that require states), state is not required
- order placed with Indian address goes to 'complete' in django admin, and state and country are included in billing address
- this takes advantage of existing functionality used for United States and Canada. I don't see any existing tests for countryStatesMap/getStates- not sure if I'm looking in the wrong place or if this doesn't pass the threshold to warrant tests?

![Screen Shot 2021-05-25 at 9 38 04 AM](https://user-images.githubusercontent.com/5384216/119507561-f02b7780-bd3c-11eb-9270-5275e2943323.png)
